drafts/regex-source-generators.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Source Code Generators and regular expressions
+domain: campzulu.hashnode.dev
+tags: architecture, .NET, C#, Blazor
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1672652289997/68d840b3-1d4d-483d-91cd-4a92a638db85.jpeg
+publishAs: linkdotnet
+ignorePost: true
+hideFromHashnodeCommunity: false
+---
+
+
+## Source Code Generators and Regular Expressions
+Since .NET 7 we have the possibility to use source code generators for regular expressions. The first question is: *What is a source code generator?*
+I can quote here [the official Microsoft page](https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview):
+> ... Source Generators let C# developers inspect user code as it is being compiled. The generator can create new C# source files on the fly that are added to the user's compilation. In this way, you have code that runs during compilation. It inspects your program to produce additional source files that are compiled together with the rest of your code.
```

Comment:
You call them "source code generators", but on the Microsoft page and here in this quote I see them only call it "Source Generators". Maybe better to use the same terminology? Or otherwise a note would be good for explanation :-)

Reply:
Very nice input. Said and done